src/Pomelo.EntityFrameworkCore.MySql/Storage/Internal/MySqlRelationalTransaction.cs

```diff
@@ -87,7 +88,7 @@ public void Dispose()
             if (!_disposed)
             {
                 _disposed = true;
-                if (_transactionOwned)
+                if (_transactionOwned && _dbTransaction.Connection.State == ConnectionState.Closed)
```

Comment:
This behavior is incorrect.  I know it's causing the outer exception, but this is not the appropriate fix.